src/main/java/nl/ordina/jobcrawler/controller/scraper/HuxleyITVacancyScraper.java

```diff
@@ -0,0 +1,146 @@
+                    .title((String)vacancyData.get("title"))
+                    .broker(BROKER)
+                    .vacancyNumber((String)vacancyData.get("jobReference"))
+                    .location((String)vacancyData.get("city"))
+                    .salary((String)vacancyData.get("salaryText"))
+                    .postingDate((String)vacancyData.get("postDate"))
+                    .about((String)vacancyData.get("description"))
+                    .build();
+
+            vacancies.add(vacancy);
+        }
+
+        return vacancies;
+    }
+
+    private HuxleyITResponse scrapeVacancies(int maxNrOfVacancies) {
+        /*
+            We are using the Spring RestTemplate for calling the HuxleyIT API POST endpoint, which provides the vacancy data.
+
+            For a successful http request we need to indicate in the http headers what type of data we are sending and what type of data we expect in return.
+            This is JSON in both cases.
+
+            The needed body for the POST request is duplicated from their frontend.
+            These parameters are noteworthy:
+            - "resultFrom": This indicates from which result number the API should start when returning results (we could use this to skip the first 50 results for example).
+            - "resultSize": This indicates the maximum number of results we want to get from the API. The API will return less results if this number is higher than the actual number of stored vacancies.
+            = "country": This indicates what country we want the returned vacancies to be located in.
+         */
+        RestTemplate restTemplate = new RestTemplate();
+
+        // Configure headers for request
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.setAccept(Collections.singletonList(MediaType.APPLICATION_JSON));
+
+        // Configure request body parameters
+        String[] country = {"Nederland"};
+        Map<String, Object> body = new HashMap<>();
+        body.put("resultSize", maxNrOfVacancies);
+        body.put("language", "nl-nl");
+        body.put("country", country);
+        body.put("brandCode", "HA");
+
+        // Build and trigger the request
+        HttpEntity<Map<String, Object>> entity = new HttpEntity<>(body, headers);
+        ResponseEntity<HuxleyITResponse> response = restTemplate.postForEntity(getSEARCH_URL(), entity, HuxleyITResponse.class);
+
+        // Return the result
+        return response.getBody();
+    }
+
+    /*
+        Because this scraper extends the abstract VacancyScraper class, it is obligated to provide a method implementation for these functions.
+        However, these methods are not used by this specific scraper, because the HuxleyIT vacancies are being read from the API directly, and not scraped from HTML.
+
+        This is why all these method implementations are empty.
```

Comment:
This is where the abstraction starts to break down. Perhaps we should rethink our model so these empty methods are not needed anymore.

This is technical debt that does not need to be resolved right away. But we need to be aware of this.

Reply:
Agreed